tcp.c

```diff
@@ -418,91 +418,52 @@ tcp_tls_get_server_pubkey(STREAM s)
 
 /* Establish a connection on the TCP layer */
 RD_BOOL
-tcp_connect(char *server)
+tcp_connect(struct addrinfo *ai)
 {
-	socklen_t option_len;
+	int i, res;
 	uint32 option_value;
-	int i;
-
-#ifdef IPv6
-
-	int n;
-	struct addrinfo hints, *res, *ressave;
-	char tcp_port_rdp_s[10];
-
-	snprintf(tcp_port_rdp_s, 10, "%d", g_tcp_port_rdp);
-
-	memset(&hints, 0, sizeof(struct addrinfo));
-	hints.ai_family = AF_UNSPEC;
-	hints.ai_socktype = SOCK_STREAM;
-
-	if ((n = getaddrinfo(server, tcp_port_rdp_s, &hints, &res)))
-	{
-		logger(Core, Error, "tcp_connect(), getaddrinfo() failed: %s", gai_strerror(n));
-		return False;
-	}
+	char hostname[64];
+	struct addrinfo *pai;
+	socklen_t option_len;
 
-	ressave = res;
+	/* connect to host */
+	pai = ai;
 	g_sock = -1;
-	while (res)
+	while (pai)
 	{
-		g_sock = socket(res->ai_family, res->ai_socktype, res->ai_protocol);
-		if (!(g_sock < 0))
+		g_sock = socket(pai->ai_family, pai->ai_socktype, pai->ai_protocol);
+		if (g_sock > 0)
 		{
-			if (connect(g_sock, res->ai_addr, res->ai_addrlen) == 0)
+			/* try connect to server */
+			res = connect(g_sock, pai->ai_addr, pai->ai_addrlen);
```

Comment:
The logging from the previous version was lost here. That would be nice to have when debugging.

Reply:
I'll move the logging for a successful connection into the loop